tests/layer_tests/tensorflow_tests/test_tf_If.py

```diff
@@ -0,0 +1,128 @@
+        assert 'y' in inputs_info, "Test error: inputs_info must contain `y`"
+        cond_shape = inputs_info['cond']
+        x_shape = inputs_info['x']
+        y_shape = inputs_info['y']
+        inputs_data = {}
+        inputs_data['cond'] = np.random.randint(0, 2, cond_shape).astype(bool)
+        inputs_data['x'] = np.random.randint(1, 10, x_shape).astype(np.float32)
+        inputs_data['y'] = np.random.randint(-50, 50, y_shape).astype(np.float32)
+        return inputs_data
+
+    def create_if_net(self, x_shape, y_shape):
+        tf.compat.v1.reset_default_graph()
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            cond = tf.compat.v1.placeholder(tf.bool, [], 'cond')
+            x = tf.compat.v1.placeholder(tf.float32, x_shape, 'x')
+            y = tf.compat.v1.placeholder(tf.float32, y_shape, 'y')
+
+            def then_branch():
+                output1 = tf.add(x, y)
+                output2 = tf.multiply(x, y)
+                output3 = tf.subtract(x, y)
+                return output1, output2, output3
+
+            def else_branch():
+                const_two = tf.constant(2.0, dtype=tf.float32)
+                output1 = tf.add(y, const_two)
+                output2 = tf.multiply(const_two, y)
+                output3 = x - y + const_two
+                return output1, output2, output3
+
+            if_output = tf.cond(cond, then_branch, else_branch)
+            tf.identity(if_output[0], name='output1')
+            tf.identity(if_output[1], name='output2')
+            tf.identity(if_output[2], name='output3')
+            tf.compat.v1.global_variables_initializer()
+
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    test_data_basic = [
+        dict(x_shape=[3], y_shape=[2, 3]),
+        dict(x_shape=[2, 1, 4], y_shape=[2, 1, 4]),
+    ]
+
+    @pytest.mark.parametrize("params", test_data_basic)
+    @pytest.mark.precommit_tf_fe
+    @pytest.mark.nightly
+    def test_if_basic(self, params, ie_device, precision, ir_version, temp_dir,
+                      use_new_frontend, use_old_api):
+        self._test(*self.create_if_net(**params),
+                   ie_device, precision, ir_version, temp_dir=temp_dir,
+                   use_new_frontend=use_new_frontend, use_old_api=use_old_api)
+
+
+class TestIfInt(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'cond' in inputs_info, "Test error: inputs_info must contain `cond`"
+        assert 'ind' in inputs_info, "Test error: inputs_info must contain `ind`"
+        assert 'y' in inputs_info, "Test error: inputs_info must contain `y`"
+        cond_shape = inputs_info['cond']
+        ind_shape = inputs_info['ind']
+        y_shape = inputs_info['y']
+        inputs_data = {}
+        inputs_data['cond'] = np.random.randint(0, 2, cond_shape).astype(bool)
+        inputs_data['ind'] = np.random.randint(1, 10, ind_shape).astype(np.int32)
```

Comment:
is the difference between these 2 tests only in the input types?
why can't we parameterize them?

Reply:
not only types, but also we have difference in bodies.

Reply:
also, in the second case there is a mix of types in inputs and outputs

Reply:
the test names are confusing in this case "TestIfInt" and "TestIfFloat". Please describe the difference in the comment or select other names.

Reply:
Overall - agree. Also I don't understand where is difference between "x" and "ind". I also see some difference between expressions, but not sure I understand are they meaningful or not.
I suggest to carefully review this part, but don't think it is merge blocker

Reply:
@itikhono, well, let me fix this in the separate PR. I think this is too minor.
Will they be fine to call as `TestIfWithFloatInputs` and `TestIfWithMixedTypes`?